modules/core/src/main/java/org/apache/ignite/internal/processors/query/RunningQueryManager.java

```diff
@@ -191,6 +193,13 @@ public void unregister(Long qryId, @Nullable Throwable failReason) {
                     canceledQrsCnt.increment();
             }
         }
+
+        Span qrySpan = qry.span();
+
+        if (failed)
+            qrySpan.addTag(ERROR, failReason::getMessage);
+
+        qrySpan.end();
```

Comment:
I'd rather use try with resources here, wrapping whole unregister() in order to guarantee that span will be closed.
Otherwise in case of any exception above qrySpan.end() we won't close the span and so that won't export it.